app/app.py

```diff
@@ -320,36 +316,42 @@ def handle_event(self, message: str):
 
         media_id = event.media_id
 
-        # Get the fragment based on the media_id
-        fragment = self._get_fragment([("dc_identifier_localid", media_id), ("IsFragment", 1)], 1)
+        # Get all items for the media id (this will include the fragment + collaterals)
+        response = self._get_fragment([("dc_identifier_localid", media_id)])
 
-        # Parse the fragment_id from the MediaHaven object
-        fragment_id = self._parse_fragment_id(fragment)
-
-        # Delete the fragment for the fragment_id
-        result = self._delete_fragment(fragment_id)
-        if not result:
+        if not response["TotalNrOfResults"]:
             raise NackException(
-                f"Unable to delete the fragment for fragment id: {fragment_id}",
-                fragment_id=fragment_id
+                f"No fragments found for media id: {media_id}",
+                media_id=media_id,
             )
 
-        self.log.info(f"Successfully deleted fragment with ID: {fragment_id}")
+        # Get each item's fragment id in a list and raise an error if there are no results.
+        fragment_ids = self._parse_fragment_ids(response["MediaDataList"])
 
-    def _parse_fragment_id(self, fragment: dict) -> str:
-        try:
-            fragment_id = fragment["MediaDataList"][0]["Internal"]["FragmentId"]
-        except KeyError as error:
-            raise NackException(
-                "FragmentId not found in the MediaHaven object",
-                error=error,
-                fragment=fragment,
+        # Delete the fragment for the fragment_id
+        for fragment_id in fragment_ids:
+            self.log.debug(
+                f"Deleting fragment for object with fragment id: {fragment_id}",
+                fragment_id=fragment_id,
+                media_id=media_id,
             )
-        return fragment_id
+            result = self._delete_fragment(fragment_id)
+            if not result:
+                raise NackException(
+                    f"Unable to delete the fragment for fragment id: {fragment_id}",
+                    fragment_id=fragment_id,
+                    media_id=media_id,
+                )
+
+        self.log.info(f"Successfully deleted item with media id: {media_id}")
+
+    def _parse_fragment_ids(self, items: List[dict]) -> List[str]:
+        fragment_ids = [item["Internal"]["FragmentId"] for item in items]
```

Comment:
Is it possible that this will raise a KeyError or will it not happen in practice?

Reply:
In that case, I think the application is allowed to blow up... ;-)